### PR TITLE
#160617638 implement rating an article

### PR DESCRIPTION
**Pull request by nakiwuge**
#### What does this PR do?
- Enables users to rate articles
#### Description of Task to be completed?
- Enable authenticated users to rate articles 
- Ensure that users do not rate articles they authored
#### How should this be manually tested?
- When you click on an article from the home page, you will be redirected to view that article. 
- Click rate this article to rate
#### What are the relevant pivotal tracker stories?
[#160617638] https://www.pivotaltracker.com/n/projects/2198902/stories/160617638
#### Screenshots 
screen shot 2018-11-16 at 12 25 41
screen shot 2018-11-16 at 12 26 08

### CODE REVIEW

File: src/routes/index.js [hunk 1]
```diff
@@ -7,18 +7,21 @@ import CreateArticle from "../components/articles/createArticle";
...
 import ResetRequest from "../components/auth/passwordReset/passwordResetPage";
 import ProtectedRoute from "./protectedRoutes";
+import Rating from "../components/articles/rating";
 
+const expressHistory = "pushState" in window.history;
 export default function AppRoutes() {
   return (
     <div>
-      <BrowserRouter>
+      <BrowserRouter forceRefresh={expressHistory}>
```
Comment (LehruAngela) [inline src/routes/index.js:18]: After merging the view an article with this task, try to do away with the forceRefresh attribute

**Review by LehruAngela — CHANGES_REQUESTED** [review]

File: src/routes/index.js (hunk 1 above)
Comment (nakiwuge) [reply src/routes/index.js:18]: @LehruAngela I have a function that should run when a user clicks on an article. In that function, I check if that user has ever rated that article or if that I user is the author of that article. The only way that function works is if I add the force refresh. I tried looking for other ways the whole of yesterday but I failed

File: src/App.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import React, { Component } from "react";
 import AppRoutes from "./routes";
 import "./assets/css/App.scss";
 import "./assets/authAssets/socialAuth.css";
+import "./assets/articleAssets/rating.scss";
```
Comment (LehruAngela) [inline src/App.js]: import both the scss files in the component that will be using it not in App.js,

File: src/__tests__/article_Tests/rating.test.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
```
Comment (LehruAngela) [inline src/__tests__/article_Tests/rating.test.js]: These tests should be in the articleTests folder, it was renamed from article_Tests to articleTests, you have two folders

File: src/actions/actionTypes.js [hunk 4]
```diff
@@ -6,6 +6,9 @@ export const PASSWORD_RESET= "PASSWORD_RESET";
 export const NEW_PASSWORD="NEW_PASSWORD";
 export const USER_NOT_FOUND="USER_NOT_FOUND";
 export const REGISTER_USER= "REGISTER_USER";
+export const RATE_ARTICLE="RATE_ARTICLE";
+export const CHECK_OWNER="CHECK_OWNER";
+export const FINISHED_RATING="FINISHED_RATING";
 const ACTION_TYPE = {
   NEW_ARTICLE: "NEW_ARTICLE",
   VIEW_ARTICLES: "VIEW_ARTICLES",
```
Comment (LehruAngela) [inline src/actions/actionTypes.js:12]: Add your actions to the ACTION_TYPE dictionary, and use the exported ACTION_TYPE to call your actions, so as to have code consistency

File: src/views/articles/createArticle.js [hunk 5]
```diff
@@ -87,7 +87,7 @@ const newArticle = ({
                             type="button"
                             id="create-button"
                             className="btn btn-sm"
-                            onSubmit={handlePublish}
+                            onClick={handlePublish}
```
Comment (LehruAngela) [inline src/views/articles/createArticle.js]: Why the change from onSubmit to onClick?

**Review by LehruAngela — CHANGES_REQUESTED** [review]

**Comment by LehruAngela** [comment]
Update the screenshots in your PR notes once you're done @zellymiriam

File: src/routes/index.js (hunk 1 above)
Comment (nakiwuge) [reply src/routes/index.js:18]: @LehruAngela I have removed this function because it failed to work after rebasing. I had to make so many changes in the View all articles component.

File: src/App.js (hunk 2 above)
Comment (nakiwuge) [reply src/App.js]: @LehruAngela I have made this change

File: src/__tests__/article_Tests/rating.test.js (hunk 3 above)
Comment (nakiwuge) [reply src/__tests__/article_Tests/rating.test.js]: @LehruAngela I have moved the tests.

File: src/actions/actionTypes.js (hunk 4 above)
Comment (nakiwuge) [reply src/actions/actionTypes.js:12]: @LehruAngela I have made this change.

File: src/views/articles/createArticle.js (hunk 5 above)
Comment (nakiwuge) [reply src/views/articles/createArticle.js]: @LehruAngela @AmosWels requested for this change since onSubmit was not working as expected.
